extensions/amp-sidebar/0.1/amp-sidebar.js

```diff
@@ -150,6 +150,15 @@ export class AmpSidebar extends AMP.BaseElement {
 
     this.action_ = Services.actionServiceForDoc(element);
 
+    if (
+      this.element.parentNode != this.element.ownerDocument.body &&
+      this.element.parentNode != this.getAmpDoc().getBody()
```

Comment:
Why is this second condition also necessary?

Reply:
For Shadow

Reply:
I was also wondering the same thing.  So in the second check, it's to check if amp-sidebar is the direct child of the body within a Shadow Dom?